packages/vite/src/node/server/ws.ts

```diff
@@ -28,7 +28,7 @@ export function createWebSocketServer(
   let httpsServer: Server | undefined = undefined
 
   const hmr = isObject(config.server.hmr) && config.server.hmr
-  const wsServer = (hmr && hmr.server) || server
+  const wsServer = (hmr && hmr.server) || (!(hmr && hmr.port) && server)
```

Comment:
Should we check for the case where `hmr.port` is the same as the main server port? We should re-use the server in that case, no?

Reply:
updated